src/rust/inetstack/mod.rs

```diff
@@ -636,7 +631,11 @@ impl<const N: usize> InetStack<N> {
     ///
     /// This function will panic if the specified future had not completed or is _background_ future.
     pub fn take_operation(&mut self, handle: TaskHandle) -> (QDesc, OperationResult) {
-        let task: OperationTask = OperationTask::from(self.scheduler.remove(handle).as_any());
+        let task: OperationTask = if let Some(task) = self.scheduler.remove(&handle) {
+            OperationTask::from(task.as_any())
+        } else {
+            panic!("Removing task that does not exist (either was previously removed or never inserted)");
```

Comment:
Shouldn't this condition be `unreachable!()`

Reply:
The scheduler doesn't have a dependency on the Demikernel fail errors. I certainly wouldn't say that this is unreachable, as we could write code to either remove the task from the scheduler more than once or accidentally call take_result in two places.

Reply:
Oh, I meant throwing the failure in the line that you are panicking. Something like:

```rust
    pub fn take_operation(&mut self, handle: TaskHandle) -> Result<(QDesc, OperationResult)> {
       if let Some(task) = self.scheduler.remove(&handle) {
             let task: OperationTask = OperationTask::from(task.as_any());
             Ok(task.get_result().expect("Coroutine not finished"))
        } else {
            Err(Fail::new(libc::EINVAL, "invalid co-routine")),
        }
    }
````

Reply:
Well, this isn't really something that we would pass back to the user because if either the coroutine didn't finish or we can't find it, then something has gone pretty wrong because we just checked the handle and it said complete. So I think panicking (and expecting) is probably the right thing here. I don't like failing with something the user won't understand. 

Either way, lets table this for now because this PR has gotten quite large.